### PR TITLE
Messaging extensions doc update for tls-registry usage

### PR DESCRIPTION
**Pull request by ozangunalp**
Another PR to add a mention in TLS registry reference page following https://github.com/quarkusio/quarkus/pull/42706

### CODE REVIEW

**Comment by gsmet** [comment]
@ozangunalp build fails with:

```
Unable to find cross reference for:
2024-09-06T15:32:57.1318313Z [ERROR] 
2024-09-06T15:32:57.1318742Z [ERROR] - messaging.adoc
2024-09-06T15:32:57.1320138Z [ERROR]     . ./tls-registry-reference.adoc#using-tls-registry in link xref:././tls-registry-reference.adoc#using-tls-registry[named TLS configuration]
```

File: docs/src/main/asciidoc/amqp-reference.adoc [hunk 1]
```diff
@@ -478,6 +478,19 @@ public AmqpClientOptions getNamedOptions() {
 }
 ----
 
+== TLS Configuration
+
+AMQP 1.0 Messaging extension integrates with the xref:./tls-registry-reference.adoc[Quarkus TLS registry] to configure the Vert.x AMQP client.
+
+To configure the TLS for an AMQP 1.0 channel, you need to provide a named TLS configuration (or `<default>`, for the default configuration) in the `application.properties`:
```
Comment (cescoffier) [inline docs/src/main/asciidoc/amqp-reference.adoc:485]: Following the ADR 0004 (https://github.com/quarkusio/quarkus/blob/main/adr/0004-using-the-tls-registry-for-clients.adoc) - it is recommended to use a named. So, I would remove the reference to `<default>`.

File: docs/src/main/asciidoc/kafka.adoc [hunk 2]
```diff
@@ -2153,6 +2153,32 @@ Update the `oauth.client.id`, `oauth.client.secret` and `oauth.token.endpoint.ur
 
 OAuth authentication works for both JVM and native modes. Since SSL in not enabled by default in native mode, `quarkus.ssl.native=true` must be added to support JaasClientOauthLoginCallbackHandler, which uses SSL. (See the xref:native-and-ssl.adoc[Using SSL with Native Executables] guide for more details.)
 
+== TLS Configuration
+
+Kafka client extension integrates with the xref:./tls-registry-reference.adoc[Quarkus TLS registry] to configure clients.
+
+To configure the TLS for the default Kafka configuration, you need to provide a named TLS configuration (or `<default>`, for the default configuration) in the `application.properties`:
```
Comment (cescoffier) [inline docs/src/main/asciidoc/kafka.adoc:2160]: Following the ADR 0004 (https://github.com/quarkusio/quarkus/blob/main/adr/0004-using-the-tls-registry-for-clients.adoc) - it is recommended to use a named. So, I would remove the reference to `<default>`.

File: docs/src/main/asciidoc/kafka.adoc [hunk 3]
```diff
@@ -2153,6 +2153,32 @@ Update the `oauth.client.id`, `oauth.client.secret` and `oauth.token.endpoint.ur
...
+----
+quarkus.tls.your-tls-config.trust-store.pem.certs=target/certs/kafka.crt,target/certs/kafka-ca.crt
+# ...
+kafka.tls-configuration-name=your-tls-config
+# enable ssl security protocol
+kafka.security.protocol=ssl
+----
+
+This will in turn provide the Kafka client with a `ssl.engine.factory.class` implementation.
+Make sure also to enable the SSL security protocol.
```
Comment (cescoffier) [inline docs/src/main/asciidoc/kafka.adoc:2172]: Can this be a callout?

File: docs/src/main/asciidoc/messaging.adoc [hunk 4]
```diff
@@ -600,6 +600,35 @@ You can disable tracing for a specific channel using the following configuration
...
+----
+
+Or you can configure it globally on all channels of a connector:
+
+[source, properties]
+----
+mp.messaging.connector.smallrye-pulsar.tls-configuration-name=my-tls-config
+----
+
+Note that for the default TLS configuration, you can provide the value `<default>`.
```
Comment (cescoffier) [inline docs/src/main/asciidoc/messaging.adoc:622]: Following the ADR 0004 (https://github.com/quarkusio/quarkus/blob/main/adr/0004-using-the-tls-registry-for-clients.adoc) - it is recommended to use a named. So, I would remove the reference to `<default>`.

File: docs/src/main/asciidoc/messaging.adoc [hunk 5]
```diff
@@ -600,6 +600,35 @@ You can disable tracing for a specific channel using the following configuration
...
+
+At the time of writing, the following messaging extensions support configuration through the Quarkus TLS Registry:
```
Comment (cescoffier) [inline docs/src/main/asciidoc/messaging.adoc:624]:
```suggestion
Currently the following messaging extensions support configuration through the Quarkus TLS Registry:
```

File: docs/src/main/asciidoc/pulsar.adoc [hunk 6]
```diff
@@ -1118,6 +1118,23 @@ class PulsarConfig {
 }
 ----
 
+==== Configuring authentication to Pulsar using mTLS
+
+Pulsar Messaging extension integrates with the xref:./tls-registry-reference.adoc[Quarkus TLS registry] to authenticate clients using mTLS.
+
+To configure the mTLS for a Pulsar channel, you need to provide a named TLS configuration (or `<default>`, for the default configuration) in the `application.properties`:
```
Comment (cescoffier) [inline docs/src/main/asciidoc/pulsar.adoc:1125]: Following the ADR 0004 (https://github.com/quarkusio/quarkus/blob/main/adr/0004-using-the-tls-registry-for-clients.adoc) - it is recommended to use a named. So, I would remove the reference to `<default>`.

File: docs/src/main/asciidoc/rabbitmq-reference.adoc [hunk 7]
```diff
@@ -389,6 +389,19 @@ You need to indicate the name of the client using the `client-options-name` attr
 mp.messaging.incoming.prices.client-options-name=my-named-options
 ----
 
+== TLS Configuration
+
+RabbitMQ Messaging extension integrates with the xref:./tls-registry-reference.adoc[Quarkus TLS registry] to configure the Vert.x RabbitMQ client.
+
+To configure the TLS for a channel, you need to provide a named TLS configuration (or `<default>`, for the default configuration) in the `application.properties`:
```
Comment (cescoffier) [inline docs/src/main/asciidoc/rabbitmq-reference.adoc:396]: Following the ADR 0004 (https://github.com/quarkusio/quarkus/blob/main/adr/0004-using-the-tls-registry-for-clients.adoc) - it is recommended to use a named. So, I would remove the reference to `<default>`.

**Review by cescoffier — CHANGES_REQUESTED** [review]

**Review by cescoffier — APPROVED** [review]

**Pull request merged**